frontend/src/app/main/ui/workspace/sidebar/options/menus/component.cljs

```diff
@@ -20,23 +21,129 @@
     [app.util.i18n :as i18n :refer [tr]]
     [rumext.v2 :as mf]))
 
-(def component-attrs [:component-id :component-file :shape-ref :main-instance?])
+(def component-attrs [:component-id :component-file :shape-ref :main-instance? :annotation])
+
+
+(mf/defc component-annotation
+  [{:keys [id values shape] :as props}]
+  (let [main-instance?        (:main-instance? values)
+        annotation            (dwah/get-main-annotation shape)
+        editing?              (mf/use-state false)
+        size                  (mf/use-state (count annotation))
+        ;; hack to create an autogrowing textarea
+        ;; based on https://css-tricks.com/the-cleanest-trick-for-autogrowing-textareas/
+        autogrow              #(let [textarea (dom/get-element "annotation-textarea")
```

Comment:
We usually use `mf/use-ref` for this, instead of `dom/get-element` (I think it's faster)

Reply:
I don't understand... I need to get the current value of the dom element. Can you give me an example?